src/main/java/com/commercetools/sync/services/impl/CartDiscountServiceImpl.java

```diff
@@ -59,6 +59,28 @@ private String getKey(@Nonnull final CartDiscountDraft cartDiscountDraft) {
         return cartDiscountQueryModel.name().lang(Locale.ENGLISH).isIn(keys);
     }
 
+    @Nonnull
+    @Override
+    public CompletionStage<Optional<String>> fetchCachedCartDiscountId(@Nonnull final String key) {
+        if (isBlank(key)) {
+            return CompletableFuture.completedFuture(Optional.empty());
+        }
+        if (keyToIdCache.containsKey(key)) {
+            return CompletableFuture.completedFuture(Optional.ofNullable(keyToIdCache.get(key)));
+        }
+        return fetchAndCache(key);
+    }
+
+    @Nonnull
+    private CompletionStage<Optional<String>> fetchAndCache(@Nonnull final String key) {
```

Comment:
Why do we need those two methods?

Reply:
Those 2 methods use can be placed as a future reference which uses caching CartDiscount IDs. Do you think this is not necessary?

Reply:
5cdd3f9de578f817bda650bce4b2d122c41f0c03